tests/css/test_stylesheet.py

```diff
@@ -101,6 +104,27 @@ def test_stylesheet_apply_empty_rulesets():
     stylesheet.apply(node)
 
 
+def test_stylesheet_apply_user_css_over_widget_css():
```

Comment:
Could we have a `Widget.CSS` with a clearly more specific rule. Something like `App .a:hover` and see if that overrides properly.

Also a rule that has `!important` within Widget.CSS, just to illustrate that can be overriden.